module/VuFind/src/VuFind/ILS/Driver/Folio.php

```diff
@@ -690,12 +690,78 @@ public function getMyTransactions($patron)
                 'id' => $this->getBibId($trans->item->instanceId),
                 'item_id' => $trans->item->id,
                 'barcode' => $trans->item->barcode,
+                'renew' => $trans->renewalCount ?? 0,
+                'renewable' => true,
                 'title' => $trans->item->title,
             ];
         }
         return $transactions;
     }
 
+    /**
+     * Get FOLIO loan IDs for use in renewMyItems.
+     *
+     * @param array $transaction An single transaction
+     * array from getMyTransactions
+     *
+     * @return string The FOLIO loan ID for this loan
+     */
+    public function getRenewDetails($transaction)
+    {
+        $loanId = $transaction['item_id'];
+        return $loanId;
+    }
+
+    /**
+     * Attempt to renew a list of items for a given patron.
+     *
+     * @param array $renewDetails An associative array with
+     * patron and details
+     *
+     * @return array $renewResult result of attempt to renew loans
+     */
+    public function renewMyItems($renewDetails)
+    {
+        $userId = $renewDetails['patron']['id'];
+        $renewalResults = [];
+
+        foreach ($renewDetails['details'] as $loanId) {
+            $renewal = [];
+            $requestbody = [
+                'itemId' => $renewDetails['details'][0],
+                'userId' => $renewDetails['patron']['id']
+                ];
+            $response = $this->makeRequest(
+                'POST', '/circulation/renew-by-id', json_encode($requestbody)
```

Comment:
I assume that for now we're stuck with renewing items one at a time, but I imagine a bulk renewal endpoint would be more efficient. Maybe something to request for the future if it doesn't already exist and/or is not already on the roadmap.

Reply:
Yes, the /circulation/renew-by-id interface takes the following loan schema as body (a string for the item id and a string for the userid https://github.com/folio-org/mod-circulation/blob/master/ramls/renew-by-id-request.json). The other renewal option is renew-by-barcode which only differs by the ID. As I understand it, the concept of a bulk renewal in the FOLIO UI is really sending several requests to one of these interfaces. I do see https://issues.folio.org/browse/CIRC-545 about implementing an interface which takes a bunch of item ids at once.